Rename unalign to LZ4_unalign in lz4.c not to clash with unalign in xxhash.c

The `lz4.c` source file contains a `typedef` of `unalign` `union` on line 346:

```c
typedef union { U16 u16; U32 u32; reg_t uArch; } __attribute__((packed)) unalign;
```

The `xxhash.c` source file contains a similar but not identical `typedef` on line 166:

```c
typedef union { U32 u32; } __attribute__((packed)) unalign;
```

If the two source files are combined (`#include`'ed) in the same translation unit (for example, to achieve better optimization without having to resort to LTO), we end up with a compiler error about conflicting declarations. For example:

```
In file included from libbutl-0.14.0/libbutl/lz4.cxx:552:
libbutl-0.14.0/libbutl/lz4.c:346:74: error: conflicting declaration 'typedef union unalign unalign'
  346 | typedef union { U16 u16; U32 u32; reg_t uArch; } __attribute__((packed)) unalign;
      |                                                                          ^~~~~~~
In file included from libbutl-0.14.0/libbutl/xxhash.h:318,
                 from libbutl-0.14.0/libbutl/lz4.cxx:537:
libbutl-0.14.0/libbutl/xxhash.c:166:52: note: previous declaration as 'typedef union unalign unalign'
  166 | typedef union { U32 u32; } __attribute__((packed)) unalign;
      |                                                    ^~~~~~~
```

I was able to fix this by renaming `unalign` in `lz4.c` to `LZ4_unalign`:

```diff
diff --git a/libbutl/lz4.c b/libbutl/lz4.c
index eac0541..3f0e430 100644
--- a/libbutl/lz4.c
+++ b/libbutl/lz4.c
@@ -343,14 +343,14 @@ static void LZ4_write32(void* memPtr, U32 value) { *(U32*)memPtr = value; }
 
 /* __pack instructions are safer, but compiler specific, hence potentially problematic for some compilers */
 /* currently only defined for gcc and icc */
-typedef union { U16 u16; U32 u32; reg_t uArch; } __attribute__((packed)) unalign;
+typedef union { U16 u16; U32 u32; reg_t uArch; } __attribute__((packed)) LZ4_unalign;
 
-static U16 LZ4_read16(const void* ptr) { return ((const unalign*)ptr)->u16; }
-static U32 LZ4_read32(const void* ptr) { return ((const unalign*)ptr)->u32; }
-static reg_t LZ4_read_ARCH(const void* ptr) { return ((const unalign*)ptr)->uArch; }
+static U16 LZ4_read16(const void* ptr) { return ((const LZ4_unalign*)ptr)->u16; }
+static U32 LZ4_read32(const void* ptr) { return ((const LZ4_unalign*)ptr)->u32; }
+static reg_t LZ4_read_ARCH(const void* ptr) { return ((const LZ4_unalign*)ptr)->uArch; }
 
-static void LZ4_write16(void* memPtr, U16 value) { ((unalign*)memPtr)->u16 = value; }
-static void LZ4_write32(void* memPtr, U32 value) { ((unalign*)memPtr)->u32 = value; }
+static void LZ4_write16(void* memPtr, U16 value) { ((LZ4_unalign*)memPtr)->u16 = value; }
+static void LZ4_write32(void* memPtr, U32 value) { ((LZ4_unalign*)memPtr)->u32 = value; }
 
 #else  /* safe and portable access using memcpy() */
 ```

--- a/lib/lz4.c
+++ b/lib/lz4.c
@@ -343,14 +343,14 @@ static void LZ4_write32(void* memPtr, U32 value) { *(U32*)memPtr = value; }
 
 /* __pack instructions are safer, but compiler specific, hence potentially problematic for some compilers */
 /* currently only defined for gcc and icc */
-typedef union { U16 u16; U32 u32; reg_t uArch; } __attribute__((packed)) unalign;
+typedef union { U16 u16; U32 u32; reg_t uArch; } __attribute__((packed)) LZ4_unalign;
 
-static U16 LZ4_read16(const void* ptr) { return ((const unalign*)ptr)->u16; }
-static U32 LZ4_read32(const void* ptr) { return ((const unalign*)ptr)->u32; }
-static reg_t LZ4_read_ARCH(const void* ptr) { return ((const unalign*)ptr)->uArch; }
+static U16 LZ4_read16(const void* ptr) { return ((const LZ4_unalign*)ptr)->u16; }
+static U32 LZ4_read32(const void* ptr) { return ((const LZ4_unalign*)ptr)->u32; }
+static reg_t LZ4_read_ARCH(const void* ptr) { return ((const LZ4_unalign*)ptr)->uArch; }
 
-static void LZ4_write16(void* memPtr, U16 value) { ((unalign*)memPtr)->u16 = value; }
-static void LZ4_write32(void* memPtr, U32 value) { ((unalign*)memPtr)->u32 = value; }
+static void LZ4_write16(void* memPtr, U16 value) { ((LZ4_unalign*)memPtr)->u16 = value; }
+static void LZ4_write32(void* memPtr, U32 value) { ((LZ4_unalign*)memPtr)->u32 = value; }
 
 #else  /* safe and portable access using memcpy() */
 

